modules/processors/multidomain_dataset_processor.py

```diff
@@ -444,3 +536,411 @@ def chunk_text(text, title, max_size=1000, overlap=200):
+            """
+            Chunks the given text into parts with a maximum size and overlap, prepending the title to each chunk.
+            
+            Args:
+            - text: The document to chunk
+            - title: document title to pre-pend to each chunk
+            - id: The id of the document
+            - max_size: Maximum size of each chunk
+            - overlap: Overlap between adjacent chunks
+            
+            Returns:
+            - A list of dictionaries with chunk 'id' and 'content' keys.
+            """
+            chunks = []
+            start = 0
+            chunk_id = 0
+            while start < len(text):
+                end = start + max_size
+                chunk = text[start:end]
+                chunk = title + ": " + chunk  # Prepend the title
+                chunks.append({'id': f"{id}_{chunk_id}", 'content': chunk})
+                start = end - overlap
+                chunk_id += 1
+
+            return chunks
+
+        all_chunks = []
+        for i in range(len(dataset)):
+            id = dataset[i]["id"]
+            doc = dataset[i]["document"]
+            assert len(doc.split(' - ')) >= 2
+            title, text = doc.split(' - ')[0], ' - '.join(doc.split(' - ')[1:])
+            chunks = chunk_text(text, title, id)
+            all_chunks.extend(chunks)
+        dataset = datasets.Dataset.from_pandas(pd.DataFrame(all_chunks).drop_duplicates(subset='content')).remove_columns(["__index_level_0__"])
+        return dataset
+
+
+class ParaphraseRC(Processor):
+    """
+    Paper: https://arxiv.org/pdf/1804.07927
+    Source: https://huggingface.co/datasets/ibm/duorc/viewer/ParaphraseRC/validation
+    DuoRC has two sub datasets: SelfRC (more direct reading comprehension) and ParaphraseRC (more challenging)
+    """
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'ParaphraseRC'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        def map_fn(row):
+            row["content"] = f"{row['title']}: {row['content']}"
+            return row
+        dataset = datasets.load_dataset("ibm/duorc", "ParaphraseRC")[self.split].filter(lambda x: not x["no_answer"]).rename_columns({"question_id":"id", "question":"content", "answers":"label"}).map(map_fn, num_proc=self.num_proc).remove_columns(["plot_id", "plot", "title", "no_answer"])
+        return dataset
+
+class ParaphraseRC_docs(Processor):
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'ParaphraseRC_docs'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        dataset = datasets.load_dataset("ibm/duorc", "ParaphraseRC")
+        if self.split == 'all':
+            dataset = datasets.concatenate_datasets([dataset["train"], dataset["validation"], dataset["test"]])
+        else:
+            dataset = dataset[self.split]
+
+        def chunk_text(text, title, id, max_size=1000, overlap=200):
```

Comment:
same comment: this looks like duplicated code ?

Reply:
done